### ISSUE TITLE
Help required with custom extension

### ISSUE THREAD

**Issue by litera** [yabwe/medium-editor#823]
I'm trying to create my own extension with its own toolbar and functionality. I've seen that some of your built-in extensions call different predefined extensions:
```js
MediumEditor.Extension.extend(...);
MediumEditor.Extension.button.extend(...);
MediumEditor.Extension.form.extend(...);
```
So I'm wondering which one should I be extending to get some of existing functionality that would be beneficial in case when an extension has to display its own toolbar?

#### Questions
1. Which one should be extended in which case?
2. Where's documentation of provided functionality of each extension?
3. Where's documentation of `MediumEditor.util` and its functionality as I've seen it used all over
4. Are there any other parts of MediumEditor object (like `util`) that extension developers can use?

**Comment by geoffliu** [comment]
Could project maintainers confirm that the APIs under `.selection` and `.util` are indeed meant to be public APIs, and they're just missing documentation? Just want to make sure we're not depending on API that's meant to be used internally.

**Comment by nmielnik** [comment]
@geoffliu Yes, we are treating this like public APIs in that we won't (intentionally) make breaking changes to them without a major release...
